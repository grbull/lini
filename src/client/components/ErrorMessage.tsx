import cn from 'classnames';
import React, { ReactElement, ReactNode } from 'react';

interface Props {
  children: string | ReactNode;
  className?: string;
  illustration?: boolean;
}

export function ErrorMessage({
  children,
  className,
  illustration,
}: Props): ReactElement {
  return (
    <div
      className={cn(
        'px-2',
        'py-2.5',
        'bg-red-400',
        'bg-opacity-50',
        'border',
        'border-solid',
        'border-red-400',
        className
      )}
    >
      {illustration && (
        <div
          className={cn('flex', 'flex-col', 'items-center', 'justify-center')}
        >
          <img
            alt="Frustrated Illustration"
            className={cn('opacity-80', 'mb-2.5')}
            src="frustrated.png"
            width={80}
          />
        </div>
      )}

      <h2 className={cn('font-bold')}>Error</h2>
      <p>{children}</p>
    </div>
  );
}

ErrorMessage.defaultProps = { className: undefined, illustration: false };
