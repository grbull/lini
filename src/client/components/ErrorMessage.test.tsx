/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import React from 'react';

import { ErrorMessage } from './ErrorMessage';

describe('ErrorMessage Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <ErrorMessage>Generic Error message</ErrorMessage>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with an illustration', () => {
    const { asFragment } = render(
      <ErrorMessage illustration>Generic Error message</ErrorMessage>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
