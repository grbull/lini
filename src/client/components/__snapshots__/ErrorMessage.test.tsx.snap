// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-2 py-2.5 bg-red-400 bg-opacity-50 border border-solid border-red-400"
  >
    <h2
      class="font-bold"
    >
      Error
    </h2>
    <p>
      Generic Error message
    </p>
  </div>
</DocumentFragment>
`;

exports[`ErrorMessage Component renders with an illustration 1`] = `
<DocumentFragment>
  <div
    class="px-2 py-2.5 bg-red-400 bg-opacity-50 border border-solid border-red-400"
  >
    <div
      class="flex flex-col items-center justify-center"
    >
      <img
        alt="Frustrated Illustration"
        class="opacity-80 mb-2.5"
        src="frustrated.png"
        width="80"
      />
    </div>
    <h2
      class="font-bold"
    >
      Error
    </h2>
    <p>
      Generic Error message
    </p>
  </div>
</DocumentFragment>
`;
